### ISSUE TITLE
REQUIREMENT: Express semantics such as classes and relationships

### ISSUE THREAD

**Issue by nishad** [dcmi/dcap#43]
Some straightforward option to notate rich semantics, which may be helpful in generating better documentation and class diagrams.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by kcoyle** [comment]
@nishad It probably will be necessary to specify what type of "classes" this refers to - for example RDF/OWL or O-O. They have considerably different semantics.

**Comment by nishad** [comment]
- 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by nishad** [comment]
Anything, which can be mapped back to RDF or any form of RDF  itself is better. Ultimately useful to generate parts of the documentation.

May be something simple like :
<img width="726" alt="image" src="https://user-images.githubusercontent.com/1372136/58365406-24c28980-7eff-11e9-8f8f-8c151c6207d0.png">

Or as complex as :
![image](https://user-images.githubusercontent.com/1372136/58365456-871b8a00-7eff-11e9-9cdf-29f0796ba3e4.png)

**Comment by kcoyle** [comment]
Yes, but note that we are not specifying the AP as RDF, although RDF can be one of the models it will support. When it comes to features that are specific to a data model (whether CSV, XML, RDF, or whatever) we may need to extend the vocabulary to include features that are not "universal". This is something we need to think about. I'll try to create an issue that will keep this in our sights.

**Comment by nishad** [comment]
The point is, If the application profile can be used to generate minimal documentation, specifically class diagrams, etc.

**Comment by kcoyle** [comment]
Actually, it would be useful to create this as a use case, please. And give as much detail as you can, along with an example. That will make the requirement more concrete, and people can extrapolate from that to their own preferred schema and documentation type. Thanks.
